Kitodo/src/main/java/org/kitodo/production/services/data/ImportService.java

```diff
@@ -1394,4 +1415,62 @@ public boolean isRecordIdentifierMetadataConfigured(RulesetManagementInterface r
     public Collection<RecordIdentifierMissingDetail> getDetailsOfRecordIdentifierMissingError() {
         return recordIdentifierMissingDetails;
     }
+
+    /**
+     * Create and return list of "SelectItem" objects for given list of "ProcessDTO"s. For each "ProcessDTO" object
+     * check whether current user can link to it as a parent process.
+     * - If a specific process belongs to a project that is not assigned to the current user, add a hint to message to
+     *   the corresponding "SelectItem"
+     * - If the user cannot link to a specific process, because the process belongs to a project which is not assigned
+     *   to him, and he also lacks the special permission to link to processes in unassigned projects, the corresponding
+     *   "SelectItem" is disabled.
+     * @param parentCandidates list of "ProcessDTO"s
+     * @param maxNumber limit
+     * @return list of "SelectItem" objects corresponding to given "ProcessDTO" objects.
+     * @throws DAOException when checking whether user can link to given "ProcessDTO"s fails
+     */
+    public ArrayList<SelectItem> getPotentialParentProcesses(List<ProcessDTO> parentCandidates, int maxNumber)
+            throws DAOException {
+        ArrayList<SelectItem> possibleParentProcesses = new ArrayList<>();
+        for (ProcessDTO process : parentCandidates.subList(0, Math.min(parentCandidates.size(), maxNumber))) {
+            SelectItem selectItem = new SelectItem(process.getId().toString(), process.getTitle());
+            selectItem.setDisabled(!userMayLinkToParent(process.getId()));
+            if (!processInAssignedProject(process.getId())) {
+                String problem = Helper.getTranslation("projectNotAssignedToCurrentUser", process.getProject().getTitle());
+                selectItem.setDescription(problem);
+                selectItem.setLabel(selectItem.getLabel() + " (" + problem + ")");
+            }
+            possibleParentProcesses.add(selectItem);
+        }
+        return possibleParentProcesses;
+    }
+
+    /**
+     * Check and return whether the process with the provided ID "processId" belongs to a project that is assigned to
+     * the current user or not.
+     * @param processId ID of the process to check
+     * @return whether the process with the provided ID belongs to a project assigned to the current user or not
+     * @throws DAOException when retrieving the process with the ID "processId" from the database fails
+     */
+    public static Boolean processInAssignedProject(int processId) throws DAOException {
```

Comment:
Return value is `false` or `true` but never `null` so return type of method can be changed to primitive `boolean`.